text/0000-named-tracers.md

```diff
@@ -0,0 +1,69 @@
+
+
+## Explanation
+
+From a user perspective, working with *Named Tracers* and *TracerFactories* is very similar to how logging frameworks like [log4j](https://www.slf4j.org/apidocs/org/slf4j/LoggerFactory.html) work. In analogy to requesting Logger objects through LoggerFactories, a tracing library would create specific 'Tracer' objects through a 'TracerFactory'.
+
+```java
+Tracer tracer = OpenTelemetry.Tracing.getTracerFactory().getTracer("io.opentelemetry.contrib.mongodb");
+```
+
+In a way, the `TracerFactory` replaces the global `Tracer` singleton object as a ubiquitous point to request a tracer instance.
+
+
+## Internal details
+
+By providing a TracerFactory and *Named Tracers*, a vendor or OpenTelemetry implementation gains more flexibility in providing tracers and can decide whether or not to produce spans for a specific traced component and even change this behavior at runtime.
+
+In the simplest case, an OpenTelemetry implementation can return a single instance for a requested tracer regardless of the name specified. This could be the case for implementations that do not want/need to enable or disable a tracer.
+
+Alternatively, an implementation can provide different tracer instances per specified tracer name, thus being able to associate this tracer with the component being traced. This allows for the possibility to enable / disable a tracer based on a custom configuration.
+
+## Trade-offs and mitigations
+
+## Prior art and alternatives
+
+Alternatively, instead of having a `TracerFactory`, existing (global) tracers could return additional indirection objects (called e.g. `TraceComponent`), which would be able to produce spans for specifically named traced components.
+
+```java
+  TraceComponent traceComponent = OpenTelemetry.Tracing.getTracer().componentBuilder("io.opentelemetry.contrib.mongodb");
+  Span span = traceComponent.spanBuilder("someMethod").startSpan();
+```
+
+Overall, this would not change a lot compared to the `TracerFactory` since the levels of indirection until producing an actual span are the same.
+
+
+## Open questions
+
+## Future possibilities
+
+By adapting this proposal, current implementations that do not honor the specified tracer name and provide a single global tracer, would not require much change. However they could change that behavior in future versions and provide more specific tracer implementations then. On the other side, if the mechanism of *Named Tracers* is not a part of the initial specification, such scenarios will be prevented and hard to retrofit in future version, should they be deemed necessary then.
+
+## Examples (for Tracer names)
+
+Since tracer names describe the libraries which use the tracers, those names should be defined in a way that makes them as unique as possible.
+
+Proposed naming is in analogy to Java package names, e.g: "io.opentelemetry.contrib.mongodb".
```

Comment:
I would remove the sole reference to Java and instead provide examples in different languages. For example, in Node.js and Python the module names are _not_ hierarchical, but the respective registries (npm, pypi) provide what could be considered canonical names.

Reply:
Thanks for the feedback. I understand how referencing Java and Java naming conventions can be misleading here. However, I would not get too hung up on syntactics here. Any (unique) string that can identify a a tracer for a given language is fine.

Also, see the comment by @discostu105 (https://github.com/open-telemetry/rfcs/pull/16#discussion_r313749544).
Would adding these examples to the proposal clarify that?

Reply:
Again, this goes to the purpose of collecting this namespace. If it's purely decorative, then sure, any string will do.

But even as decorative, using a Resource rather than a string could make more sense, because resources have attributes, e.g. I might want to know not just that this span came from library X, but from version N of library X.

Reply:
I really prefer using a string as a key here, since this is a simple and already established (see the conversation on logging frameworks) pattern.

Also, there is a proposal https://github.com/open-telemetry/oteps/pull/26/files which intends to move Resources from API into SDK, which does not make them a good candidate for a key here.

However, adding the version information you are looking for, is currently addressed in another proposal https://github.com/open-telemetry/opentelemetry-specification/issues/235 (which should probably go into oteps as well).